ecopod/superUSDC.md

```diff
@@ -0,0 +1,129 @@
+
+**Separate bridge contracts for every OP Chain**
+
+Originally proposed for this work, an alternative approach involves each OP Chain separate and independent lockbox/bridge contract instead of coupling to a shared L1 bridge contract with other OP Chains. This method allows each chain to fully own the bridge and token contracts during the initial stages before coordinating the migration to native USDC.
+
+While this approach is simpler and less architecturally complex, it is not future-proof for the Superchain ecosystem. It makes it challenging to achieve important objectives such as liquidity unification and full compatibility with Interop. This alternative may be suitable for individual OP Chains that do not intend to integrate into the Superchain ecosystem.
+
+# Proposed Solution
+
+> 📑 *A high level overview of the proposed solution. When there are multiple alternatives there should be an explanation of why one solution was picked over other solutions. As a rule of thumb, including code snippets (except for defining an external API) is likely too low level.*
+
+
+Recreation of the flows for each stage of the implementation of this design. Note that each flow is unique for each OP chain, except for contract deployment on L1, which is done once and remains common throughout the Superchain.
+
+### 1) L1→L2 Deployments
+
+Getting the factory, upgrade-manager, bridge-adapter and the implementations of the L2 adapter and token on L1. This is necessary to securely deploy on existing and new chains:
+
+1. Anyone can deploy `SuperUSDCFactory`, `UpgradeManager`, the Proxy Adapter, and the initial implementation for the Adapter on L1. The `UpgradeManager` owns the Proxy Adapter on L1.
+2. Anyone set up the Trusted 3rd Party as the owner of the `UpgradeManager`, which can be done at deployment or via a change owner function.
+3. The Trusted 3rd Party whitelists `CrossDomainMessenger` addresses in the `UpgradeManager`. There are two possible flows:
+    1. Allow users to suggest addresses and let the Trusted 3rd Party approve them.
+    2. The Trusted 3rd Party sets addresses directly.
+4. Anyone can initialize a deployment to a chain with a whitelisted `CrossDomainMessenger` by calling the `SuperUSDCFactory` contract.
+    1. The `SuperUSDCFactory` calls the `CrossDomainMessenger` with the deployment calldata.
+    2. The calldata is used to deploy and set up the L2 contracts: Proxy Adapter, Implementation Adapter, Proxy `SuperUSDC`, and Implementation `SuperUSDC`.
```

Comment:
Is the `CrossDomainMessenger` able to deploy L2 contracts pre-deterministically?  Reason I'm asking is because if these tokes are the same address across op-stack, it would be simple to atomically bridge across supported Superchains.  If these tokens have different addresses, there would be an additional requirement to wrap/unwrap when jumping across Superchains.